lib/ukio_web/controllers/booking_controller.ex

```diff
@@ -8,11 +8,22 @@ defmodule UkioWeb.BookingController do
   action_fallback UkioWeb.FallbackController
 
   def create(conn, %{"booking" => booking_params}) do
-    with {:ok, %Booking{} = booking} <- BookingCreator.create(booking_params) do
-      conn
-      |> put_status(:created)
-      |> render(:show, booking: booking)
+    case BookingCreator.create(booking_params) do
+      {:ok, %Booking{} = booking} ->
+        conn
+        |> put_status(:created)
+        |> render(:show, booking: booking)
+
+      {:error, :apartment_not_available} ->
+        conn
+        |> put_status(:unauthorized)
+        |> json(%{error: "The apartment is not available for the specified dates."})
     end
+  rescue
+    _ ->
+      conn
+      |> put_status(:unprocessable_entity)
+      |> json(%{errors: %{base: "Invalid booking params"}})
```

Comment:
if any other error is thrown by `BookingCreator.create()`, for example, non existing apartment, `rescue` will catch that error and return a `422`